### ISSUE TITLE
Add PrimeFaces p:selectOneRadio component to Visual Editor

### ISSUE THREAD

**Issue by piotrzarzycki21** [prominic/Moonshine-IDE#375]
More information about -> [p:selectOneRadio](https://www.primefaces.org/showcase/ui/input/oneRadio.xhtml)

**Comment by piotrzarzycki21** [comment]
I'm reviewing your code and here is what should be added:
1. `columns` - should have API description, usage and default value is 0, not 3.
1. since you are using `Grid` as a base of that component - please add property `layout` and in property editor drop down list with two values `lineDirection`, `grid` where `lineDirection` is default
1. columns is active only if `layout` - grid is selected
1. I think component should have property `value`

In case of Grid we have two options of generating it:
```
<p:selectOneRadio id="city" value="#{radioView.city}"  layout="grid" columns="3">
            <f:selectItems value="#{radioView.cities}" var="c" itemLabel="#{city}" itemValue="#{city}"/>
</p:selectOneRadio>
```
And:
```
<p:selectOneRadio layout="grid" columns="3">
			<f:selectItem itemLabel="Red" itemValue="Red" />
            <f:selectItem itemLabel="Green" itemValue="Green" />
            <f:selectItem itemLabel="Blue" itemValue="Blue" />
			<f:selectItem itemLabel="Red" itemValue="Red" />
            <f:selectItem itemLabel="Green" itemValue="Green" />
            <f:selectItem itemLabel="Blue" itemValue="Blue" />
			<f:selectItem itemLabel="Red" itemValue="Red" />
            <f:selectItem itemLabel="Green" itemValue="Green" />
            <f:selectItem itemLabel="Blue" itemValue="Blue" />
</p:selectOneRadio>
```

Not sure what output.

**Comment by piotrzarzycki21** [comment]
Do not forget adding in the component description how Visual Editor XML will looks like. Paste it here as an  example, cause I may miss it in committed code.

**Comment by piotrzarzycki21** [comment]
Yes I was thinking that I went a bit early, but It was better to do this on this stage of the code and you have at least some kind of specification.

**Comment by piotrzarzycki21** [comment]
> I was in mind to provide some way to choose between Grid or basic SelectRadioButton usage

Do you mean here using as it is in example some other control `p:radioButton` ?

**Comment by JoelProminic** [comment]
I did some testing with the history on the list entries:
- Adding entries
- Editing entries
- Deleting entries

Undo/Redo seemed to work fine in all of these cases.

**Comment by piotrzarzycki21** [comment]
@rat-moonshine You need to file documentation for that component.

**Comment by rat-moonshine** [comment]
I'm not very good with documentation.. but I tried. I have updated documentation to all three of these components - TextEditor, SelectOneRadio and SelectOneMenu. @piotrzarzycki21 check if they looks legit to you, point me if needs some amendments. 

Thanks!

**Comment by piotrzarzycki21** [comment]
TextEditor and SelectOneMenu looks good to me. I did corrections some time ago in TextEditor. The most important thing is on the top of the classes - follow how documentation looks like in other components. 

Next - each property which is visible in property editor should be documented. If it's not in property editor it's rarely appear in documentation. There are exceptions like constants `ELEMENT_NAME` or `PRIME_FACES_XML_ELEMENT_NAME`. 

I see also that in almost all components you were using `Exclude `tag on top of `isSelected` - It won't work. Tag `Exclude `have to be used on top of the class. If you would like to exclude some property/method from comments you need to use:
```
/**
   * @exclude
   *
**/
```

However I found that above doesn't work in each case, so better use `Exclude`.

**Comment by rat-moonshine** [comment]
Nice! Did you checked SelectOneRadio, too? By the way, I have replaced all the `@exclude` usage with simple `Exclude` words. 

Is the URL handy to you to the API docs for PrimeFaces components?

**Comment by piotrzarzycki21** [comment]
If you would like to check how your documentation looks like you need to build VisualEditor using ant script. It will generate documentation in bin/doc.

I did check, but I see that SelectOneMenu is also have bad documentation on top of the class.
I will reopen #373 with description. 

SelectOneRadio have following right now:
```
Visual Editor XML:
     <SelectOneRadio
     Attributes
     width="230"
     height="21"
	 value=""
     percentWidth=""
	 layout=""
	 columns=""
     percentHeight="">
	 <selectItem itemLabel="Title" itemValue="" itemVar="" value=""/>
	 />
     
PrimeFaces output:
     <p:selectOneRadio
     Attributes
	 value=""
	 layout=""
	 columns=""
     style="width:230px;height:21px;">
	 	<f:selectItem itemLabel="Title" itemValue="" itemVar="" value=""/>
     />
```

The question is what will happen if you copy paste PrimeFaces output to real PrimeFaces project - Will it work? Do you see issue above ? 
What will happen if you copy/paste Visual Editor XML ? - It will be translated correctly ? 
What is the columns default ? It it's zero it should be in above examples.

**Comment by rat-moonshine** [comment]
@piotrzarzycki21 thanks for explaining this in details. 

That 'Attributes' word - that looks a bit odd in the described tag - is that suppose to be there? Although, I see that word declared against each components' tag description. 

```
<p:selectOneRadio
     Attributes
	 value=""
```

I also see the tags above malformed. I shall look into these. 

Do you have the URL to the online APIDocs to the PrimeFaces components?

**Comment by piotrzarzycki21** [comment]
I do have - it's among the emails, but I don't have ability to update it whenever it changes. You need to build project by ant and check how your documentation looks like. Once it will be ready we need to ask Andrii for update.

**Comment by piotrzarzycki21** [comment]
Word 'Attributes' is just bold element. Looks like that in generated documentation:

![attr_tag](https://user-images.githubusercontent.com/24554795/46067643-9d5a2600-c177-11e8-98b3-2f122e6756d3.png)

**Comment by rat-moonshine** [comment]
I was not talking about the APIDocs that we produce (which is at here anyway: http://moonshine-ide.com/documentation/primefaces/), but the some other original PrimeFaces APIDocs that you gave me some time back (I'm having difficulty to find it now..)

**Comment by piotrzarzycki21** [comment]
Ahh that one! Here you go [PrimeFaces 6.2](https://www.primefaces.org/docs/guide/primefaces_user_guide_6_2.pdf)

**Comment by rat-moonshine** [comment]
Okay, I have updated the tag description as this, check now: 

```
<SelectOneRadio
     Attributes
     width="230"
     height="21"
     value=""
     columns="0"
     percentWidth=""
     percentHeight="">
	 <selectItem itemLabel="Title" itemValue="" itemVar="" value=""/>
    </SelectOneRadio>
     
PrimeFaces output: 
    <p:selectOneRadio
     Attributes
     value=""
     columns="0"
     style="width:230px;height:21px;">
	<f:selectItem itemLabel="Title" itemValue=""/>
     </p:selectOneRadio>
```

**Comment by piotrzarzycki21** [comment]
It looks good to me, however why in PrimeFaces output there is no `itemVar` and `value` in `selectItem `?
In public properties there is not width/height and percentWidht/percentHeight. 
![selectoneradio](https://user-images.githubusercontent.com/24554795/46077745-3e55da80-c192-11e8-82e1-337dead5d2aa.png)

**Comment by rat-moonshine** [comment]
`itemVar` and `value` actually part of _selectItems_ (not _selectItem_). We have those fields available to PrimeFaces/Moonshine XML so we can use them in future. Generated PrimeFaces code will only generate `itemLabel` and `itemValue` for now.

I have updated the documentation with introducing width/height and percentWidth/percentHeight as _public_ properties now.
